calibanGenClient generates invalid code

Consider a call to `calibanGenClient`:

```
calibanGenClient https://shopify.dev/admin-graphql-direct-proxy/2025-04 shopify.scala
```

It will generate the code and `scalafmt` fails with an error:
```
error: Client.scala: org.scalafmt.dynamic.exceptions.PositionExceptionImpl: Client.scala:64653: error: [dialect scala3] `)` expected but `integer constant` found
fulfillmentOrdersOptIn : scala.Option[Boolean] = None, @deprecated("The non-SKU sharing fulfillment services are now deprecated. As of API version "2025-10" this argument will return an error if set to false.")
                                                                                                                                                    ^
Caused by: Client.scala:64653: error: [dialect scala3] `)` expected but `integer constant` found
fulfillmentOrdersOptIn : scala.Option[Boolean] = None, @deprecated("The non-SKU sharing fulfillment services are now deprecated. As of API version "2025-10" this argument will return an error if set to false.")
                                                                                                                                                    ^
	at org.scalafmt.Scalafmt$.$anonfun$doFormatOne$2(Scalafmt.scala:150)
	at scala.meta.parsers.Parsed.fold(Errors.scala:12)
	at scala.meta.parsers.Parsed.fold$(Errors.scala:10)
	at scala.meta.parsers.Parsed$Error.fold(Errors.scala:28)
	at org.scalafmt.Scalafmt$.doFormatOne(Scalafmt.scala:153)
	at org.scalafmt.Scalafmt$.doFormat(Scalafmt.scala:127)
	at org.scalafmt.Scalafmt$.formatCodeWithStyle(Scalafmt.scala:108)
	at org.scalafmt.Scalafmt$.$anonfun$formatCode$3(Scalafmt.scala:81)
	at scala.util.Success.fold(Try.scala:271)
	at org.scalafmt.Scalafmt$.formatCode(Scalafmt.scala:81)
	at org.scalafmt.Scalafmt$.format(Scalafmt.scala:60)
	at org.scalafmt.Scalafmt.format(Scalafmt.scala)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.scalafmt.dynamic.ScalafmtReflect.$anonfun$tryFormat$1(ScalafmtReflect.scala:121)
	at scala.util.Try$.apply(Try.scala:213)
	at org.scalafmt.dynamic.ScalafmtReflect.tryFormat(ScalafmtReflect.scala:117)
	at org.scalafmt.dynamic.ScalafmtReflectConfig.tryFormat(ScalafmtReflectConfig.scala:95)
	at org.scalafmt.dynamic.ScalafmtDynamicSession.$anonfun$tryForceFormat$1(ScalafmtDynamicSession.scala:49)
	at scala.util.Success.flatMap(Try.scala:251)
	at org.scalafmt.dynamic.ScalafmtDynamicSession.tryForceFormat(ScalafmtDynamicSession.scala:49)
	at org.scalafmt.dynamic.ScalafmtDynamicSession.tryFormat(ScalafmtDynamicSession.scala:36)
	at org.scalafmt.dynamic.ScalafmtDynamicSession.format(ScalafmtDynamicSession.scala:18)
	at org.scalafmt.dynamic.ScalafmtDynamic.format(ScalafmtDynamic.scala:52)
	at caliban.tools.Formatter$.$anonfun$format$5(Formatter.scala:40)
	at scala.collection.immutable.List.map(List.scala:293)
	at caliban.tools.Formatter$.$anonfun$format$2(Formatter.scala:40)
	at zio.ZIOCompanionVersionSpecific.$anonfun$attempt$1(ZIOCompanionVersionSpecific.scala:100)
	at zio.ZIO$.$anonfun$suspendSucceed$1(ZIO.scala:4867)
	at zio.internal.FiberRuntime.runLoop(FiberRuntime.scala:1063)
	at zio.internal.FiberRuntime.runLoop(FiberRuntime.scala:1152)
	at zio.internal.FiberRuntime.runLoop(FiberRuntime.scala:1094)
	at zio.internal.FiberRuntime.evaluateEffect(FiberRuntime.scala:412)
	at zio.internal.FiberRuntime.evaluateMessageWhileSuspended(FiberRuntime.scala:487)
	at zio.internal.FiberRuntime.drainQueueOnCurrentThread(FiberRuntime.scala:249)
	at zio.internal.FiberRuntime.run(FiberRuntime.scala:137)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:829)
```

Which is because of the following generated:
```
@deprecated("The non-SKU sharing fulfillment services are now deprecated. As of API version "2025-10" this argument will return an error if set to false.")
```
Obviously, this should have escaped the inner quotes:
```
@deprecated("The non-SKU sharing fulfillment services are now deprecated. As of API version \"2025-10\" this argument will return an error if set to false.")
```

--- a/tools/src/main/scala/caliban/tools/ClientWriter.scala
+++ b/tools/src/main/scala/caliban/tools/ClientWriter.scala
@@ -768,15 +768,19 @@ object ClientWriter {
         case _               => ""
       }
 
-    def writeDeprecated(reason: Option[String]): String =
+    def writeDeprecated(reason: Option[String]): String = {
+      def escapeDoubleQuotes(input: String): String =
+        input.replace("\"", "\\\"")
+
       reason match {
         case None                              =>
           "@deprecated\n"
         case Some(body) if body.contains("\n") =>
-          s"@deprecated($tripleQuotes$body$tripleQuotes)\n"
+          s"@deprecated($tripleQuotes${escapeDoubleQuotes(body)}$tripleQuotes)\n"
         case Some(body)                        =>
-          s"@deprecated($doubleQuotes$body$doubleQuotes)\n"
+          s"@deprecated($doubleQuotes${escapeDoubleQuotes(body)}$doubleQuotes)\n"
       }
+    }
 
     def writeType(t: Type): String = t match {
       case NamedType(name, true)   => safeTypeName(name)

--- a/tools/src/test/scala/caliban/tools/ClientWriterSpec.scala
+++ b/tools/src/test/scala/caliban/tools/ClientWriterSpec.scala
@@ -449,6 +449,24 @@ object ClientWriterSpec extends SnapshotTest {
                maybeMaybeOptionList: [Option]
              }
             """)
+      },
+      test("deprecation reason with quotes should be properly escaped") {
+        val schema = """
+             type Character {
+               name: String! @deprecated(reason: "The non-SKU sharing fulfillment services are now deprecated. As of API version \"2025-10\" this argument will return an error if set to false.")
+             }
+            """
+
+        gen(schema).map { result =>
+          // The generated code should contain escaped quotes
+          assertTrue(result.contains("\\\"2025-10\\\""))
+          // The generated code should not contain unescaped quotes that would break Scala syntax
+          assertTrue(
+            !result.contains(
+              "@deprecated(\"The non-SKU sharing fulfillment services are now deprecated. As of API version \"2025-10\" this argument will return an error if set to false.\")"
+            )
+          )
+        }
       }
     ) @@ TestAspect.parallelN(4)
 }

